src/pages/task/taskTabs/TestsTable.tsx

```diff
@@ -221,44 +225,47 @@ const getInitialState = (queryParams: {
 };
 
 const getQueryVariables = (
-  search: string,
+  queryParams: { [key: string]: any },
   taskId: string,
 ): TaskTestsQueryVariables => {
-  const parsed = parseQueryString(search);
-
   // Detemining sort category
-  const parsedCategory = (parsed[RequiredQueryParams.Category] ?? "")
-    .toString()
-    .toUpperCase();
-  const TestSortCategories = Object.keys(TestSortCategory).map(
-    (k) => TestSortCategory[k],
-  );
-  const sortBy = TestSortCategories.includes(parsedCategory)
-    ? (parsedCategory as TestSortCategory)
+  const parsedSortBy = getString(queryParams[TableQueryParams.SortBy]);
+  const testSortCategories: string[] = Object.values(TestSortCategory);
+  const sortBy = testSortCategories.includes(parsedSortBy)
+    ? (parsedSortBy as TestSortCategory)
     : undefined;
+  const sorts = queryParams[TableQueryParams.Sorts];
 
   // Determining sort direction
-  const parsedDirection = (parsed[RequiredQueryParams.Sort] ?? "").toString();
+  const parsedDirection = getString(queryParams[TableQueryParams.SortDir]);
   const direction =
     parsedDirection === SortDirection.Desc
       ? SortDirection.Desc
       : SortDirection.Asc;
 
-  const sort = sortBy && direction ? [{ sortBy, direction }] : [];
+  let sort = [];
+  if (sortBy && direction) {
+    sort = [{ sortBy, direction }];
+  } else if (sorts) {
+    sort = parseSortString(sorts, {
+      sortByKey: "sortBy",
+      sortDirKey: "direction",
+    });
+  }
```

Comment:
Is it possible to consolidate the test table to just use one format of sorting? I think it's a little confusing how the sorting format changes based on the number of sorts applied. Or would it cause too many problems with the other tables like the hosts table?

Reply:
I looked into this too but sadly think it would be very cumbersome. Particularly due to the "sort by" field; different tables have different values that would be valid there.

Reply:
That's okay!! 😔 Could we consider a ticket to consolidate the sorts across our tables (not for this epic, but for future work)?

Reply:
DEVPROD-4236